mineral-ui/mineral-ui
# [tokens]: Extract and distribute design tokens

### Description

- Add mineral-ui-tokens package
    - Extract source tokens from default theme
        - Rename `placeholder_color` to `input_color_placeholder`
        - Append `px` units to `fontSize_base` and `breakpoint_[width]` variables
        - Change `font_family` value from `Open Sans` to `"Open Sans"`
    - Build scripts based on mineral-ui-icons
    - Generate output files (so they can be reliably referenced within mineral-ui)
- Refactor styles & themes to consume the generated token files
- Remove `colors` export and directory
- Misc project & site updates
    - Update .gitignore to ignore tokens debug output
    - Update FauxControl with renamed theme variable (`input_color_placeholder`)
    - Always render an appropriate border on color values in variable tables
    - Refactor of ControlPanel and Theming/index
    - Replace all imports of `colors` with `palette` from mineral-ui-tokens
    - Move tokens.js from themes to website and rename to groupedMineralTheme.js
- Update snapshots

### Motivation and context

Closes #664 

### Screenshots, videos, or demo, if appropriate


https://644-tokens--mineral-ui.netlify.com/

### How to test


1. `cd mineral-ui && npm start`
2. There should be _no_ visual changes to the site, except the changes detailed above that are visible in the [theme variable table](https://644-tokens--mineral-ui.netlify.com/theming#common-scenarios-theme-structure)
3. `cd packages/mineral-ui-tokens`
4. Run all scripts in package.json and confirm no errors
5. Check generated files for correctness
6. Check site and components, which now reference those generated files

### Types of changes

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

**Breaking**

1. The `placeholder_color` theme variable was renamed to `input_color_placeholder`.
2. Color ramps in the new package are sparse arrays, rather than objects like they were in `colors`, e.g. `blue.blue_60` is now accessed as `blue[60]`.
3. `px` units were added to `fontSize_base` and `breakpoint_[width]` theme variables, making them strings rather than numbers

### Checklist

* [x] Automated tests written and passing _Existing tests pass_



### How does this PR make you feel?

![Mario tokens](https://media.giphy.com/media/GNvWw0pDL6QRW/giphy.gif)
